pkg/app/piped/cloudprovider/kubernetes/applier.go

```diff
@@ -35,6 +35,8 @@ type Applier interface {
 	ReplaceManifest(ctx context.Context, manifest Manifest) error
 	// Delete deletes the given resource from Kubernetes cluster.
 	Delete(ctx context.Context, key ResourceKey) error
+	// GetManifest gets the manifest by the given resource.
+	GetManifest(ctx context.Context, key ResourceKey) (Manifest, error)
```

Comment:
How about instead of adding a new `GetManifest` like this, we add the validating logic inside the `Delete` method?
Because I feel that `Applier` should not have any read method.

Reply:
Exactly, that makes sense.
That's why let me apply it.